public/app/core/components/sidemenu/SignIn.tsx

```diff
@@ -1,16 +1,18 @@
 import React, { FC } from 'react';
+import config from 'app/core/config';
 import { connectWithStore } from 'app/core/utils/connectWithReduxStore';
 import { StoreState } from 'app/types';
 import { Icon } from '@grafana/ui';
 
-export const SignIn: FC<any> = ({ url }) => {
-  const splitted = url.split('?');
-  let params = new URLSearchParams('');
-  if (splitted.length > 1) {
-    params = new URLSearchParams(splitted[1]);
-  }
+const getForcedLoginUrl = (url: string) => {
+  const urlObj = new URL(url, config.appUrl);
```

Comment:
Out of curiosity what will be the difference between `appUrl` and `appSubUrl`?

Reply:
`appSubUrl` is only the subpath URL part (if exists) which is better illustrated in [these](https://github.com/grafana/grafana/blob/cf2343fac82706bdb409ccd6401585f6f81fb8d1/pkg/setting/setting_test.go#L310) examples.

Reply:
thank you @papagian